devito/mpi/distributed.py

```diff
@@ -90,26 +97,47 @@ def dimensions(self):
     @property
     def neighbours(self):
         """
-        Return the mapper ``proc -> direction``; ``proc`` is the rank of a
-        neighboring process, while ``direction`` tells whether ``proc`` is
+        Return the mapper ``proc -> side``; ``proc`` is the rank of a
+        neighboring process, while ``side`` tells whether ``proc`` is
         logically at right (value=1) or left (value=-1) of ``self``.
         """
         shifts = {d: self._comm.Shift(i, 1) for i, d in enumerate(self.dimensions)}
         ret = {}
         for d, (src, dest) in shifts.items():
             ret[d] = {}
-            ret[d][LEFT] = src if src != -1 else None
-            ret[d][RIGHT] = dest if dest != -1 else None
+            ret[d][LEFT] = src
+            ret[d][RIGHT] = dest
         return ret
 
-    def __repr__(self):
-        return "Distributor(nprocs=%d)" % self.nprocs
-
+    @cached_property
+    def _C_comm(self):
+        """
+        A :class:`Object` wrapping an MPI communicator.
 
-class RankRelativePosition(Tag):
-    pass
+        Extracted from: ::
 
+            https://github.com/mpi4py/mpi4py/blob/master/demo/wrap-ctypes/helloworld.py
+        """
+        from devito.types import CompositeObject
```

Comment:
Import here?

Reply:
there's a cyclic dependency that I haven't fixed yet, which appears if I move this to the top of the file. But this code is evolving rapidly, so..